apisix/plugins/proxy-rewrite.lua

```diff
@@ -178,6 +183,31 @@ function _M.rewrite(conf, ctx)
         ctx.var.upstream_uri = upstream_uri
     end
 
+    -- enum    = {"GET", "POST","PUT",""PATCH,"DELETE"}
+    -- change HTTP method
+    if conf.method ~=nil then
+
+      if conf.method == "GET" then
+        ngx.req.set_method(ngx.HTTP_GET)
+
+      elseif conf.method == "POST" then
+        ngx.req.set_method(ngx.HTTP_POST)
+
+      elseif conf.method == "PUT" then
+        ngx.req.set_method(ngx.HTTP_PUT)
+                
+      elseif conf.method == "PATCH" then
+        ngx.req.set_method(ngx.HTTP_PATCH)
+
+      elseif conf.method == "DELETE" then
+        ngx.req.set_method(ngx.HTTP_DELETE)
+      else
+        core.log.warn("Invalid HTTP method: ", conf.method)
+      end
+
+      core.log.info("HTTP method changed to: ", conf.method)
```

Comment:
I think the original HTTP method  also should be printed.

```suggestion
      core.log.info("changed HTTP method from ", old_method, " to ", conf.method)
```